app/src/main/java/org/ea/sqrl/utils/PasswordStrengthMeter.java

```diff
@@ -0,0 +1,214 @@
+            else result.rating = PasswordRating.GOOD;
+
+            // Last chance to detect cancellation
+            if (isCancelled()) return null;
+
+            return result;
+        }
+
+        @Override
+        protected void onPostExecute(PasswordStrengthResult result) {
+
+            if (result == null) return;
+
+            try {
+                Drawable ledGreen = ContextCompat.getDrawable(mContext, R.drawable.led_green);
+                Drawable ledRed   = ContextCompat.getDrawable(mContext, R.drawable.led_red);
+                TextView txtPasswordStrength = mPassStrengthLayout.findViewById(R.id.txtPasswordStrength);
+                TextView txtPasswordWarning = mPassStrengthLayout.findViewById(R.id.txtPasswordWarning);
+                ImageView imgUppercase = mPassStrengthLayout.findViewById(R.id.imgPasswordContainsUppercase);
+                ImageView imgLowercase = mPassStrengthLayout.findViewById(R.id.imgPasswordContainsLowercase);
+                ImageView imgDigits = mPassStrengthLayout.findViewById(R.id.imgPasswordContainsDigits);
+                ImageView imgSymbols = mPassStrengthLayout.findViewById(R.id.imgSymbols);
+                ProgressBar progressPwStrength = mPassStrengthLayout.findViewById(R.id.progressPasswordStrength);
+
+                progressPwStrength.setMax(STRENGTH_POINTS_MIN_GOOD);
+                progressPwStrength.setProgress(result.strengthPoints);
+
+                if (result.rating == PasswordRating.POOR) {
+                    txtPasswordStrength.setText(mContext.getText(R.string.password_strength_poor));
+                    txtPasswordStrength.setBackgroundColor(
+                            ContextCompat.getColor(mContext, R.color.password_strength_poor));
+                }
+                else if (result.rating == PasswordRating.MEDIUM) {
+                    txtPasswordStrength.setText(mContext.getText(R.string.password_strength_medium));
+                    txtPasswordStrength.setBackgroundColor(
+                            ContextCompat.getColor(mContext, R.color.password_strength_medium));
+                }
+                else if (result.rating == PasswordRating.GOOD) {
+                    txtPasswordStrength.setText(mContext.getText(R.string.password_strength_good));
+                    txtPasswordStrength.setBackgroundColor(
+                            ContextCompat.getColor(mContext, R.color.password_strength_good));
+                }
+
+                if (result.lowercaseUsed) imgLowercase.setImageDrawable(ledGreen);
+                else imgLowercase.setImageDrawable(ledRed);
+
+                if (result.uppercaseUsed) imgUppercase.setImageDrawable(ledGreen);
+                else imgUppercase.setImageDrawable(ledRed);
+
+                if (result.digitsUsed) imgDigits.setImageDrawable(ledGreen);
+                else imgDigits.setImageDrawable(ledRed);
```

Comment:
Again to keep to the coding standard I would either use brackets around the statements or use the online select like 

```result.digitsUsed ? ledGreen : ledRed```

Reply:
Point taken.